File: Daily Challanges/2026-06-June/2026-06-28.py
def connect_three(board):
    if not board or not board[0]:
        return []

    rows, cols = len(board), len(board[0])
    directions = [(0, 1), (1, 0), (1, 1), (1, -1)]

    for r in range(rows):
        for c in range(cols):
            if board[r][c] == "":
                continue
            for dr, dc in directions:
                cells = []
                for k in range(3):
                    nr = r + dr * k
                    nc = c + dc * k
                    if not (0 <= nr < rows and 0 <= nc < cols):
                        break
                    if board[nr][nc] != board[r][c]:
                        break
                    cells.append([nr, nc])
                if len(cells) == 3:
                    cells.sort()
                    return [board[r][c]] + cells
    return []

print(connect_three([
    ["", "", "", ""],
    ["", "", "", ""],
    ["", "Y", "", ""],
    ["Y", "R", "R", "R"]
]))

print(connect_three([
    ["", "", "", ""],
    ["", "Y", "Y", ""],
    ["", "Y", "R", "R"],
    ["", "Y", "R", "R"]
]))

print(connect_three([
    ["", "", "Y", "R"],
    ["", "Y", "R", "Y"],
    ["", "R", "Y", "R"],
    ["", "R", "Y", "R"]
]))

print(connect_three([
    ["", "Y", "", ""],
    ["", "Y", "Y", ""],
    ["", "R", "R", "Y"],
    ["R", "R", "Y", "R"]
]))

print(connect_three([
    ["Y", "R", "R", "Y"],
    ["R", "Y", "Y", "R"],
    ["Y", "R", "R", "Y"],
    ["R", "Y", "Y", "R"]
]))

File: Daily Challanges/2026-06-June/2026-06-28.js
function connectThree(board) {
  if (!board.length || !board[0].length) return [];

  const rows = board.length;
  const cols = board[0].length;
  const directions = [
    [0, 1],
    [1, 0],
    [1, 1],
    [1, -1],
  ];

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (board[r][c] === "") continue;

      for (const [dr, dc] of directions) {
        const cells = [];

        for (let k = 0; k < 3; k++) {
          const nr = r + dr * k;
          const nc = c + dc * k;

          if (nr < 0 || nr >= rows || nc < 0 || nc >= cols) break;
          if (board[nr][nc] !== board[r][c]) break;

          cells.push([nr, nc]);
        }

        if (cells.length === 3) {
          cells.sort((a, b) => a[0] - b[0] || a[1] - b[1]);
          return [board[r][c], ...cells];
        }
      }
    }
  }

  return [];
}

console.log(
  connectThree([
    ["", "", "", ""],
    ["", "", "", ""],
    ["", "Y", "", ""],
    ["Y", "R", "R", "R"],
  ]),
);

console.log(
  connectThree([
    ["", "", "", ""],
    ["", "Y", "Y", ""],
    ["", "Y", "R", "R"],
    ["", "Y", "R", "R"],
  ]),
);

console.log(
  connectThree([
    ["", "", "Y", "R"],
    ["", "Y", "R", "Y"],
    ["", "R", "Y", "R"],
    ["", "R", "Y", "R"],
  ]),
);

console.log(
  connectThree([
    ["", "Y", "", ""],
    ["", "Y", "Y", ""],
    ["", "R", "R", "Y"],
    ["R", "R", "Y", "R"],
  ]),
);

console.log(
  connectThree([
    ["Y", "R", "R", "Y"],
    ["R", "Y", "Y", "R"],
    ["Y", "R", "R", "Y"],
    ["R", "Y", "Y", "R"],
  ]),
);
